src/mixins/controls.js

```diff
@@ -173,14 +200,12 @@ const OrbitControlsMixin = (superclass) => class extends superclass {
         this.render();
     }
 
-    showSecondAxes() {
-        const secondAxes = [this.scene2.x, this.scene2.y, this.scene2.z].filter(x => x);  // assert no `undefined`;
-        this.scene2.add(...secondAxes);
-    }
 
     hideSecondAxes() {
-        const secondAxes = [this.scene2.x, this.scene2.y, this.scene2.z].filter(x => x);  // assert no `undefined`;
-        this.scene2.remove(...secondAxes);
+        this.scene2.remove(this.scene2.secondAxesGroup);
+        while (this.renderer2CSS2D.domElement.firstChild) {
+            this.renderer2CSS2D.domElement.removeChild(this.renderer2CSS2D.domElement.firstChild);
+        }
```

Comment:
I don't understand why we need the while loop also - this is a potential performance concern. Let's just organize all objects in groups and remove groups directly